### PR TITLE
[3party] Optionally allow building with packaged expat

### PR DESCRIPTION
**Pull request by Ferenc-**
Signed-off-by: Ferenc Géczi <ferenc.gm@gmail.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -297,7 +297,12 @@ set(EXPAT_BUILD_DOCS OFF)
 set(EXPAT_BUILD_PKGCONFIG OFF)
 set(EXPAT_ENABLE_INSTALL OFF)
 set(EXPAT_SHARED_LIBS OFF)
-add_subdirectory(3party/expat/expat)
+include_directories("${PROJECT_BINARY_DIR}/3party/expat/expat/lib")
```
Comment (biodranik) [inline CMakeLists.txt:300]: Why is this line needed?

File: coding/internal/xmlparser.hpp [hunk 2]
```diff
@@ -13,7 +13,7 @@
 #include <string>
 
 #define XML_STATIC
-#include "3party/expat/expat/lib/expat.h"
+#include "expat.h"
```
Comment (biodranik) [inline coding/internal/xmlparser.hpp:16]: Should it be `<expat.h>`?

Comment (Ferenc-) [reply coding/internal/xmlparser.hpp]: Yes. Fixed.

**Comment by Ferenc-** [comment]
@biodranik the review comments are all fixed. And now that the OSM test API is available again all tests have passed.

File: CMakeLists.txt [hunk 3]
```diff
@@ -297,7 +297,11 @@ set(EXPAT_BUILD_DOCS OFF)
 set(EXPAT_BUILD_PKGCONFIG OFF)
 set(EXPAT_ENABLE_INSTALL OFF)
 set(EXPAT_SHARED_LIBS OFF)
-add_subdirectory(3party/expat/expat)
+find_package_or_fallback_to_3party(
+  "expat"
+  "2.5.0"
+  "3party/expat/expat")
```
Comment (biodranik) [inline CMakeLists.txt:303]: nit: do it in one line. Fewer lines = easier to read/overview code.

**Review by biodranik — APPROVED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (Ferenc-) [reply CMakeLists.txt]: Done

**Comment by Ferenc-** [comment]
@biodranik is there anything else needed here?

**Review by biodranik — APPROVED** [review]

**Comment by biodranik** [comment]
Thanks!
